packages/@aws-cdk/aws-events-targets/lib/lambda.ts

```diff
@@ -37,9 +37,10 @@ export class LambdaFunction implements events.IRuleTarget {
         sourceArn: rule.ruleArn
       });
     }
+    const id = this.handler.node.uniqueId;
 
     return {
-      id: this.handler.node.id,
+      id: id.substring(Math.max(id.length - 64, 0), id.length),
```

Comment:
Might I suggest a slightly different solution:

- Just return `uniqueId` here. Please make the same change for all rule targets at once, I don't really see a point in having a single PR for each one.
- Add the substring operation here: https://github.com/awslabs/aws-cdk/blob/master/packages/@aws-cdk/aws-events/lib/rule.ts#L233 , it will be automatically applied to all rule targets.

Reply:
Ok, got it: you mean pass the `uniqueId` from outside *as is* and the `sanitizeId()` will cut it accordingly to be compliant with `AWS::Events::Rule Target` specification. Makes perfect sense, thank you for your feedback :)